Handler/Sites.hs

```diff
@@ -102,11 +102,19 @@ siteForm msite = renderBootstrap3 BootstrapBasicForm $ Site
 
     baseUrlField :: Field Handler Text
     baseUrlField = flip checkM textField $ \url -> do
-        n <- runDB $ count [SiteBaseUrl ==. url]
+        if unchanged url
+            then return $ Right url
+            else ensureUnique url
 
-        return $ case n of
-            0 -> Right url
-            _ -> Left ("This URL is already in use" :: Text)
+      where
+        unchanged url = fmap siteBaseUrl msite == Just url
```

Comment:
Functions in a `where` clause have access to the enclosing function's arguments. This is the `msite` argument passed to `siteForm`.